postinstall.js

```diff
@@ -1,26 +1,135 @@
-if (path_1.dirname(packageJsonPath) !== "nativescript-dev-appium") {
-    package_json_helper_1.updatePackageJsonDep(packageJsonPath, appRootPath, isTscProj, true);
+
+const { existsSync, mkdirSync, readdirSync, readFileSync, statSync, writeFileSync } = require("fs");
+const { basename, resolve } = require("path");
+
+const appRootPath = require('app-root-path').toString();
+const childProcess = require("child_process");
+const e2eTests = "e2e";
+const e2eProjectFolderPath = resolve(appRootPath, e2eTests);
+const e2ePluginFolderPath = resolve(appRootPath, "node_modules", "nativescript-dev-appium", "e2e");
+const packageJsonPath = resolve(appRootPath, "package.json");
+const packageJson = JSON.parse(readFileSync(packageJsonPath, "utf8"));
+const isTypeScriptProject =
+    (
+        packageJson.dependencies &&
+        packageJson.dependencies.hasOwnProperty("typescript")
+    ) || (
+        packageJson.devDependencies &&
+        packageJson.devDependencies.hasOwnProperty("typescript")
+    );
+const isWin = /^win/.test(process.platform);
+
+if (basename(appRootPath) !== "nativescript-dev-appium") {
+    updatePackageJsonDependencies(packageJson, isTypeScriptProject);
 }
-if (!utils_1.fileExists(e2eLocalFolderPath)) {
-    fs_1.mkdirSync(e2eLocalFolderPath);
-    if (isTscProj) {
-        console.log("TypeScript project: adding a test sample ...");
-        utils_1.copy(e2ePluginFolderPath, e2eLocalFolderPath, true);
+
+if (!existsSync(e2eProjectFolderPath)) {
+    mkdirSync(e2eProjectFolderPath);
+    if (isTypeScriptProject) {
+        console.info("TypeScript project - adding a sample test ...");
+        copy(e2ePluginFolderPath, e2eProjectFolderPath);
+    } else {
+        console.info("JavaScript project - not adding a sample test ...");
     }
-    else {
-        console.log("JavaScript project: not adding a test sample ...");
+}
+
+function copy(src, dest) {
+    if (!existsSync(src)) {
+        return Error("Source doesn't exist: " + src);
+    }
+    if (statSync(src).isDirectory()) {
+        if (!existsSync(dest)) {
+            mkdirSync(dest);
+        }
+
+        const entries = new Array();
+        readdirSync(resolve(src)).forEach(entry => {
+            entries.push(entry);
+        });
+
+        entries.forEach(entry => {
+            const source = resolve(src,  entry);
+            const destination = resolve(dest, entry);
+            console.info("Copying " + source + " to " + destination + " ...");
+            copy(source, destination);
+        });
+    } else {
+        writeFileSync(dest, readFileSync(src));
     }
 }
+
+function executeNpmInstall(cwd) {
+    let spawnArgs = [];
+    let command = "";
+    if (isWin) {
+        command = "cmd.exe"
+        spawnArgs = ["/c", "npm", "install"];
+    } else {
+        command = "npm"
+        spawnArgs = ["install"];
+    }
+    const npm = childProcess.spawnSync(command, spawnArgs, { cwd: cwd, stdio: "inherit" });
```

Comment:
You don't need to assign that to a variable.
Also `cwd: cwd` can be shortened to just `cwd`